src/plugins/data/public/shard_failure_modal/shard_failure_description.tsx

```diff
@@ -41,25 +48,84 @@ export function formatValueByKey(value: unknown, key: string): string | JSX.Elem
 }
 
 export function ShardFailureDescription(props: ShardFailure) {
+  const [showDetails, setShowDetails] = useState<boolean>(false);
+
   const flattendReason = getFlattenedObject(props.reason);
 
-  const listItems = Object.entries(flattendReason).map(([key, value]) => ({
-    title: formatKey(key),
-    description: formatValueByKey(value, key),
-  }));
+  const reasonItems = Object.entries(flattendReason)
+    .filter(([key]) => key !== 'type')
```

Comment:
QQ: What's the reason with the key 'type' to be filtered out?

Reply:
@kertal In the design which Andrea suggested `Node` is between `Type` and `Reason`.

<img width="300" alt="Screenshot 2023-07-04 at 09 47 29" src="https://github.com/elastic/kibana/assets/1415710/cf9d2d5e-92d5-414c-a332-dcef0fc4c2d5">

We could also move `Type` after `Node` then this code would not be necessary. Wdyt?

Reply:
Thx, I see. Well, I might be wrong, but seeing the type of the error might be of higher interest than the node at the first sight, so I'm fine keeping it that way